shiny/render/_try_render_plot.py

```diff
@@ -15,15 +15,113 @@
     from matplotlib.figure import Figure
 
 
+class PlotSizeInfo:
+    """This class carries information from the render.plot transformer to the logic that
+    actually renders the plot to PNG. It also encapsulates the tricky logic for figuring
+    out what the image size and attributes should be."""
+
+    def __init__(
+        self,
+        container_size_px_fn: tuple[Callable[[], float], Callable[[], float]],
+        user_specified_size_px: tuple[float | None, float | None],
+        pixelratio: float,
+    ):
+        """
+        Args:
+            container_size_px_fn: A tuple of two functions that return the width and
+                height of the container, in pixels. If the user specified an explicit
+                width/height on the @render.plot decorator, then these functions should
+                not be called, as doing so will take a reactive dependency on that
+                dimension.
+            user_specified_size_px: A tuple of two floats, or None. If the user
+                specified an explicit width/height on the @render.plot decorator, then
+                the corresponding float will be that value. Otherwise, it will be None.
+            pixelratio: The device pixel ratio that was detected from the client.
+        """
+        self._container_size_px_fn = container_size_px_fn
+        self.user_specified_size_px = user_specified_size_px
+        self.pixelratio = pixelratio
+
+    def get_img_size_px(
+        self,
+        fig_initial_size_inches: tuple[float, float],
+        fig_result_size_inches: tuple[float, float],
+        dpi: float,
+    ) -> tuple[float, float, str, str]:
+        """
+        Determines the desired size of the image in logical pixels (not doubling
+        resolution for retina displays--the caller needs to worry about that), and the
+        width and height attributes that should be set on the <img> tag.
+
+        Args:
+            fig_initial_size_inches: The size of the figure before the user's plotting
```

Comment:
Please make clear that "figure" refers to a matplotlib figure, and not img tag in the DOM.